### ISSUE TITLE
[$40]Submissions Tab: submissions table issues

### ISSUE THREAD

**Issue by SathyaJayabal** [topcoder-platform/community-app#4552]
V4 URL: https://www.topcoder.com/challenges/30127007?tab=submissions
V5  URL: https://beta-community-app.topcoder.com/challenges/6b219116-4aac-4bb0-8a9d-6ec76e2d4c24?tab=submissions

1. When there are more than one submission for a user, we must display the user only once.

2. User rating colour must be reflected on the handle and rating. #4530 

3. Initial Final Scores not displayed
<img width="1440" alt="Screenshot 2020-06-17 at 6 33 28 PM" src="https://user-images.githubusercontent.com/58783823/84901589-53427200-b0c9-11ea-9d8e-be3d6d6a7285.png">
<img width="1440" alt="Screenshot 2020-06-17 at 6 35 40 PM" src="https://user-images.githubusercontent.com/58783823/84901632-62c1bb00-b0c9-11ea-8991-2bfd883d8549.png">

4. For F2F and bug hunt challenges the rating column is not shown in Prod, but it is displayed on V5.
@Oanh-and-only-Oanh . please confirm if this is indeed an issue in  V5 and we need to remove the rating column for F2F challenges.
<img width="1440" alt="Screenshot 2020-06-17 at 6 41 18 PM" src="https://user-images.githubusercontent.com/58783823/84902266-38bcc880-b0ca-11ea-9b21-389301ec27b6.png">
<img width="1440" alt="Screenshot 2020-06-17 at 6 41 28 PM" src="https://user-images.githubusercontent.com/58783823/84902285-41ad9a00-b0ca-11ea-8045-f1757d21e108.png">

**Comment by Oanh-and-only-Oanh** [comment]
@SathyaJayabal, we will leave the ratings in for QA and f2f

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30130313 has been created for this ticket.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30130313 has been updated - it has been assigned to sr_jr.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by simranb86** [comment]
@Sushil what all type of challenges we looking for here :point_up:

**Comment by simranb86** [comment]
Also in /submissions api response, in review I see some have 2 like below and some have only 1 and some none. So what goes as Initial and Final Score ?

0: {score: 100, updatedBy: "MeaSS8yPG4VIer4XalVqOAYhnCNecGfN@clients",…}
1: {score: 100, updatedBy: "maE2maBSv9fRVHjSlC31LFZSq6VhhZqC@clients",…}

**Comment by simranb86** [comment]
Also for `User rating colour must be reflected on the handle and rating.` - where do we get color styles? Can you share style/rating mapping ? 

@sushilshinde ☝️

**Comment by sushilshinde** [comment]
@SathyaJayabal @Oanh-and-only-Oanh

**Comment by simranb86** [comment]
@sushilshinde I am done with 1 and 2, I need confirmation on how to map scores before I push PR. 

@Oanh-and-only-Oanh @SathyaJayabal can you confirm ☝️ ?

**Comment by sushilshinde** [comment]
@simranb86 skip item # 2 as it will be implemented separately 

and PR

**Comment by simranb86** [comment]
> @simranb86 skip item # 2 as it will be implemented separately
> 
> and PR

@sushilshinde I have already implemented # 1 and # 2, I am waiting for information on how to map reviews to complete # 3

**Comment by SathyaJayabal** [comment]
@simranb86 , rating colour details
https://github.com/topcoder-platform/community-app/issues/4289#issuecomment-619785058

**Comment by simranb86** [comment]
> @simranb86 , rating colour details
> [#4289 (comment)](https://github.com/topcoder-platform/community-app/issues/4289#issuecomment-619785058)

Thanks @SathyaJayabal will update the PR with the new color codes. 

@Oanh-and-only-Oanh @sushilshinde I am still waiting for confirmation on how to map Initial and Final Reviews from response. Can one of you please help so we can take this ticket ahead ?

**Comment by SathyaJayabal** [comment]
> @sushilshinde what all type of challenges we looking for here ☝️ - Code and F2F only right ?

@simranb86 , for BH and F2F , the rating column is displayed in V5. It was hidden in V4. But no changes needed as we have decided to keep the rating columns for F2F and BH.(https://github.com/topcoder-platform/community-app/issues/4552#issuecomment-645608619)

**Comment by SathyaJayabal** [comment]
> Also in /submissions api response, in review I see some have 2 like below and some have only 1 and some none. So what goes as Initial and Final Score ?
> 
> ```
> 0: {score: 100, updatedBy: "MeaSS8yPG4VIer4XalVqOAYhnCNecGfN@clients",…}
> created: "2020-04-10T05:57:23.710Z"
> createdBy: "MeaSS8yPG4VIer4XalVqOAYhnCNecGfN@clients"
> id: "203ab03e-9e79-4657-a5b7-b70c0b0f668c"
> reviewerId: "d1282f60-c97e-4da3-83fc-04ec5e262f09"
> score: 100
> scoreCardId: 51827436
> status: "completed"
> submissionId: "0e1d51e4-f7e5-4cd3-918a-275f2df41ec4"
> typeId: "d96d5f17-5884-47b8-bfea-bddf066e451f"
> updated: "2020-04-10T05:57:23.710Z"
> updatedBy: "MeaSS8yPG4VIer4XalVqOAYhnCNecGfN@clients"
> 1: {score: 100, updatedBy: "maE2maBSv9fRVHjSlC31LFZSq6VhhZqC@clients",…}
> created: "2020-04-10T05:57:16.514Z"
> createdBy: "maE2maBSv9fRVHjSlC31LFZSq6VhhZqC@clients"
> id: "ea679c09-3e5d-4cc4-bd27-a84dec085c75"
> reviewerId: "af941e9b-6c7b-4623-968c-218341f43b92"
> score: 100
> scoreCardId: 30001850
> status: "completed"
> submissionId: "0e1d51e4-f7e5-4cd3-918a-275f2df41ec4"
> typeId: "68c5a381-c8ab-48af-92a7-7a869a4ee6c3"
> updated: "2020-04-10T05:57:16.514Z"
> updatedBy: "maE2maBSv9fRVHjSlC31LFZSq6VhhZqC@clients"
> ```

@simranb86 , the typeId in the response refers to the type of review. For example typeId "68c5a381-c8ab-48af-92a7-7a869a4ee6c3" may refer to virus scan review, etc. 
We will have to map the initial and final scores to the corresponding  typeIds. @sushilshinde , please help with mapping.

**Comment by SathyaJayabal** [comment]
@simranb86 , noticed incorrect submission count for this one challenge
https://beta-community-app.topcoder.com/challenges/ad26ade8-aa6f-4623-bb65-ac510c70bef3?tab=submissions
<img width="1440" alt="Screenshot 2020-06-23 at 6 20 58 PM" src="https://user-images.githubusercontent.com/58783823/85405701-6f895780-b57e-11ea-9bbe-f7e9690831b0.png">

**Comment by simranb86** [comment]
> > Also in /submissions api response, in review I see some have 2 like below and some have only 1 and some none. So what goes as Initial and Final Score ?
> > ```
> > 0: {score: 100, updatedBy: "MeaSS8yPG4VIer4XalVqOAYhnCNecGfN@clients",…}
> > created: "2020-04-10T05:57:23.710Z"
> > createdBy: "MeaSS8yPG4VIer4XalVqOAYhnCNecGfN@clients"
> > id: "203ab03e-9e79-4657-a5b7-b70c0b0f668c"
> > reviewerId: "d1282f60-c97e-4da3-83fc-04ec5e262f09"
> > score: 100
> > scoreCardId: 51827436
> > status: "completed"
> > submissionId: "0e1d51e4-f7e5-4cd3-918a-275f2df41ec4"
> > typeId: "d96d5f17-5884-47b8-bfea-bddf066e451f"
> > updated: "2020-04-10T05:57:23.710Z"
> > updatedBy: "MeaSS8yPG4VIer4XalVqOAYhnCNecGfN@clients"
> > 1: {score: 100, updatedBy: "maE2maBSv9fRVHjSlC31LFZSq6VhhZqC@clients",…}
> > created: "2020-04-10T05:57:16.514Z"
> > createdBy: "maE2maBSv9fRVHjSlC31LFZSq6VhhZqC@clients"
> > id: "ea679c09-3e5d-4cc4-bd27-a84dec085c75"
> > reviewerId: "af941e9b-6c7b-4623-968c-218341f43b92"
> > score: 100
> > scoreCardId: 30001850
> > status: "completed"
> > submissionId: "0e1d51e4-f7e5-4cd3-918a-275f2df41ec4"
> > typeId: "68c5a381-c8ab-48af-92a7-7a869a4ee6c3"
> > updated: "2020-04-10T05:57:16.514Z"
> > updatedBy: "maE2maBSv9fRVHjSlC31LFZSq6VhhZqC@clients"
> > ```
> 
> @simranb86 , the typeId in the response refers to the type of review. For example typeId "68c5a381-c8ab-48af-92a7-7a869a4ee6c3" may refer to virus scan review, etc.
> We will have to map the initial and final scores to the corresponding typeIds. @sushilshinde , please help with mapping.

@SathyaJayabal @sushilshinde can you help with mapping api if there is one ? Since others are done, I will close this once I hear back on mapping.

**Comment by SathyaJayabal** [comment]
@lakshmiathreya , can you help getting the mapping details for @simranb86 . What reviewTypeIds will map to the initial and final scores that is displayed on the submissions table on the challenge details page.

**Comment by simranb86** [comment]
@sushilshinde it has been like more than a week now, should we take off scoring part and address it as separate issue ? I have the other 2 in place in PR shared already.

**Comment by simranb86** [comment]
1 & 2 are already completed here - topcoder-platform/topcoder-react-lib#202

**Comment by SathyaJayabal** [comment]
@simranb86 , lets see if Mithun can help
https://github.com/topcoder-platform/submissions-api/issues/186

**Comment by sushilshinde** [comment]
Sorry about this we were focusing on P0s and P1s

**Comment by sushilshinde** [comment]
@rootelement https://github.com/topcoder-platform/submissions-api/issues/186#issuecomment-652952077

**Comment by SathyaJayabal** [comment]
@simranb86 , we are tracking the score issue (3) in a separate ticket.
https://github.com/topcoder-platform/community-app/issues/4627

**Comment by simranb86** [comment]
PR https://github.com/topcoder-platform/topcoder-react-lib/pull/202/

**Comment by sushilshinde** [comment]
@simranb86 tests are failing

https://circleci.com/gh/topcoder-platform/topcoder-react-lib/1108

**Comment by luizrrodrigues** [comment]
@sushilshinde My latest PR to `topcoder-react-lib` fix that.
https://github.com/topcoder-platform/topcoder-react-lib/pull/208

**Comment by sushilshinde** [comment]
they thanks @luizrrodrigues

**Comment by SathyaJayabal** [comment]
@simranb86 @sushilshinde ,  Dont see the the submissions on the  submissions table for all challenges.
<img width="1440" alt="Screenshot 2020-07-16 at 9 55 21 AM" src="https://user-images.githubusercontent.com/58783823/87626723-ba942600-c74a-11ea-8055-dc5f75f6877e.png">

<img width="1440" alt="Screenshot 2020-07-16 at 9 54 57 AM" src="https://user-images.githubusercontent.com/58783823/87626643-7bfe6b80-c74a-11ea-9f70-9650d554d6da.png">

**Comment by SathyaJayabal** [comment]
Raising the priority.

**Comment by SathyaJayabal** [comment]
This fix was reverted. Submissions are visible now. Reducing the priority.
<img width="1440" alt="Screenshot 2020-07-17 at 1 17 26 PM" src="https://user-images.githubusercontent.com/58783823/87761928-e76e3900-c82f-11ea-9095-cd4783a52d41.png">

**Comment by SathyaJayabal** [comment]
https://github.com/topcoder-platform/community-app/issues/4552#issue-640432688
Only 1 and 2 in scope for this ticket

cc @sushilshinde

**Comment by sushilshinde** [comment]
@topcoder-platform/topcodercompetitors @topcoder-platform/tcxcommunity this is open for pickup

Submit in 8 hours and PR to

https://github.com/topcoder-platform/community-app/tree/integration-v5-challenge-api
https://github.com/topcoder-platform/topcoder-react-lib/tree/integration-v5-challenge-api

run `npm run test` before PR

**Comment by crazyk07** [comment]
Contest https://www.topcoder.com/challenges/30130313 has been updated - it has been assigned to PkDurlabhji.<br/><br/>```This is an automated message for crazyk via Topcoder X```

**Comment by sushilshinde** [comment]
@PrakashDurlabhji any updates?

**Comment by PrakashDurlabhji** [comment]
@sushilshinde apology for delay, in few hours it will be completed. rating color is changed by me. working for duplicates

**Comment by SathyaJayabal** [comment]
@PrakashDurlabhji , please hold off on working on the duplicates.

**Comment by PrakashDurlabhji** [comment]
@SathyaJayabal ok then raising the MR

**Comment by PrakashDurlabhji** [comment]
@sushilshinde @SathyaJayabal done in https://github.com/topcoder-platform/community-app/pull/4670

**Comment by SathyaJayabal** [comment]
verified on beta
<img width="1440" alt="Screenshot 2020-07-27 at 11 23 20 AM" src="https://user-images.githubusercontent.com/58783823/88508116-a29a8d00-cffb-11ea-9340-642ce391d475.png">

**Comment by crazyk07** [comment]
Payment task has been updated: https://software.topcoder.com/review/actions/ViewProjectDetails?pid=30130313<br/><br/>```This is an automated message for crazyk via Topcoder X```
